Composer/packages/server/src/controllers/provision.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import axios from 'axios';
+import { ExtensionContext } from '@bfc/extension';
+
+import { BotProjectService } from '../services/project';
+
+const defaultPublishConfig = {
+  name: 'default',
+  type: 'localpublish',
+  provisionConfig: '',
+  configuration: JSON.stringify({}),
+};
+
+export const ProvisionController = {
+  // get all subscriptions for current user
+  getSubscriptions: async (req, res) => {
+    if (!req.headers || !req.headers.authorization) {
+      res.status(401).json({ message: 'Unauthorized' });
+      return;
+    }
+    // extract the token from the authorization header, where it is in the form `Bearer <TOKEN>`
+    // we pass this directly on to Azure.
+    const authHeader = req.headers.authorization;
+
+    try {
+      const result = await axios.get('https://management.azure.com/subscriptions?api-version=2020-01-01', {
+        headers: { Authorization: authHeader },
+      });
+      res.status(result.status).json(result.data);
+    } catch (err) {
+      res.status(err.response.status).json(err.response.data);
+    }
+  },
+  getResourceGroups: async (req, res) => {
+    if (!req.headers || !req.headers.authorization) {
+      res.status(401).json({ message: 'Unauthorized' });
+      return;
+    }
+    // extract the token from the authorization header, where it is in the form `Bearer <TOKEN>`
+    // we pass this directly on to Azure.
+    const authHeader = req.headers.authorization;
+
+    const result = await axios.get(
```

Comment:
* All of these API calls to external services need some error handling.  At a minimum, please implement the try/catch seen in getSubscriptions. However, consider what are the likely error scenarios and whether passing the error directly back to the client is the appropriate behavior.

For example, we know that each one of these may throw an authorization error.  Is a special response necessary for this?

Another possible known issue is that there is a network failure or timeout on the API call.  

What are other common error scenarios for each of these APIs? The official documentation for them likely includes some information.  

* In addition, each one of these should include inline documentation linking to the official API docs, explaining what each function is doing, and also documenting the expected output format.